### PR TITLE
SONARIAC-1540 S7021: WORKDIR instruction should only be used with absolute path

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

File: iac-extensions/docker/src/main/resources/org/sonar/l10n/docker/rules/docker/S7021.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<p><code>WORKDIR</code> instruction should be used with absolute path for clarity and reliability.</p>
```
Comment (jonas-wielage-sonarsource) [inline iac-extensions/docker/src/main/resources/org/sonar/l10n/docker/rules/docker/S7021.html:1]: IMO `instructions` and adding an `an` before absolute path would sound a bit better

File: iac-extensions/docker/src/test/resources/checks/WorkdirAbsolutPathCheck/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
+FROM scratch
+
+WORKDIR /absolute/path/unix
+WORKDIR "/absolute/path/unix"
+
```
Comment (jonas-wielage-sonarsource) [inline iac-extensions/docker/src/test/resources/checks/WorkdirAbsolutPathCheck/Dockerfile:5]: `WORKDIR ""` would be non compliant, is this expected?

**Review by jonas-wielage-sonarsource — APPROVED** [review]
LGTM, please have a look at my suggestions before merging

File: iac-extensions/docker/src/test/resources/checks/WorkdirAbsolutPathCheck/Dockerfile (hunk 2 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/test/resources/checks/WorkdirAbsolutPathCheck/Dockerfile:5]: Indeed, even if it's a bit odd, doesn't reflect well if something like that appear on SonarLint. I will handle this use case.

File: iac-extensions/docker/src/main/resources/org/sonar/l10n/docker/rules/docker/S7021.html (hunk 1 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/main/resources/org/sonar/l10n/docker/rules/docker/S7021.html:1]: Good catch! Here is the [PR](https://github.com/SonarSource/rspec/pull/4091) on RSPEC.

**Pull request merged**